### PR TITLE
is_list property for FletcherDtype

### PR DESCRIPTION
**Pull request by marc9595**

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -195,6 +195,10 @@ def _is_numeric(self):
             or pa.types.is_decimal(self.arrow_dtype)
         )
 
+    @property
+    def _is_list(self):
```
Comment (krivonogov) [inline fletcher/base.py]: we will use it as is? `dtype._is_list` (in which case I prefer name without underscore) or there is some pandas method relying on `_is_list`

Comment (marc9595) [reply fletcher/base.py]: The convention for `_is_numeric` and `_is_boolean` in pandas is to write them with an underscore (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.extensions.ExtensionDtype.html). So I followed the same convention, but I agree that it should be a public property.

Comment (krivonogov) [reply fletcher/base.py]: but they don't define `_is_list`, so I would use `is_list`

Comment (marc9595) [reply fletcher/base.py]: That's ok with me

Comment (marc9595) [reply fletcher/base.py]: @artemru Is it okay for you to have `is_list` instead of `_is_list` ?

**Pull request merged**